primitive-types/impls/serde/src/serialize.rs

```diff
@@ -88,32 +89,27 @@ impl fmt::Display for FromHexError {
 	}
 }
 
-/// Decode given hex string into a vector of bytes.
+/// Decode given (both 0x-prefixed or not) hex string into a vector of bytes.
 ///
-/// Returns an error if the string is not prefixed with `0x`
-/// or non-hex characters are present.
+/// Returns an error if non-hex characters are present.
 pub fn from_hex(v: &str) -> Result<Vec<u8>, FromHexError> {
-	if !v.starts_with("0x") {
-		return Err(FromHexError::MissingPrefix)
-	}
+	let (v, stripped) = v.strip_prefix("0x").map_or((v, false), |v| (v, true));
 
-	let mut bytes = vec![0u8; (v.len() - 1) / 2];
-	from_hex_raw(v, &mut bytes)?;
+	let mut bytes = vec![0u8; (v.len() + 1) / 2];
```

Comment:
dq: is `(v.len() + 1) / 2` used to work in case the hex string is not divisible by 2?

when it's even we would allocate one extra byte but yeah it was like this before too..

Reply:
yeah, there're some tests like
```rust
assert_eq!(from_hex("102"), Ok(vec![1, 2]));
```
which wouldn't pass if `bytes.len()` would be 1 instead of 2, so we need to round up. It was `- 1` previously because we were always expecting `0x` prefix to present.